src/Init/Data/BitVec/Bitblast.lean

```diff
@@ -198,4 +198,53 @@ theorem ule_eq_not_ult (x y : BitVec w) : x.ule y = !y.ult x := by
 theorem ule_eq_carry (x y : BitVec w) : x.ule y = carry w y (~~~x) true := by
   simp [ule_eq_not_ult, ult_eq_not_carry]
 
+/-- If two bitvectors have the same `msb`, then signed and unsigned comparisons coincide -/
+theorem slt_eq_ult_of_msb_eq {x y : BitVec w} (h : x.msb = y.msb) :
+    x.slt y = x.ult y := by
+  simp only [BitVec.slt, toInt_eq_msb_cond, BitVec.ult, decide_eq_decide, h]
+  cases y.msb <;> simp only [↓reduceIte, Int.ofNat_lt, Int.sub_lt_sub_right_iff]
+
+/-- If two bitvectors have different `msb`s, then unsigned comparison is determined by this bit -/
+theorem ult_eq_msb_of_msb_neq {x y : BitVec w} (h : x.msb ≠ y.msb) :
+    x.ult y = y.msb := by
+  simp [BitVec.ult]
+  cases hy : y.msb <;> simp only [decide_eq_true_eq, decide_eq_false_iff_not, Nat.not_lt, ge_iff_le]
+  · have hx : x.msb = true := by simp_all
+    have := (msb_eq_false_iff_two_mul_lt _).mp hy
+    have := (msb_eq_true_iff_two_mul_ge _).mp hx
+    omega
+  · have hx : x.msb = false := by simp_all
+    have := (msb_eq_false_iff_two_mul_lt _).mp hx
+    have := (msb_eq_true_iff_two_mul_ge _).mp hy
+    omega
+
+/-- If two bitvectors have different `msb`s, then signed and unsigned comparisons are opposites -/
+theorem slt_eq_not_ult_of_msb_neq {x y : BitVec w} (h : x.msb ≠ y.msb) :
+    x.slt y = !x.ult y := by
+  have hx : x.msb = !y.msb := by revert h; cases x.msb <;> cases y.msb <;> trivial
+  simp only [BitVec.slt, toInt_eq_msb_cond, hx, not_eq_true', ult_eq_msb_of_msb_neq h]
+  cases y.msb
+  <;> simp only [↓reduceIte, false_eq_true, Bool.not_false, decide_eq_true_eq, gt_iff_lt,
+        Bool.not_true, decide_eq_false_iff_not]
+  <;> omega
```

Comment:
I would be satisfied with
```
  simp only [BitVec.slt, toInt_eq_msb_cond, hx, ult_eq_msb_of_msb_neq h]
  cases y.msb <;> (simp; omega)
```
here (i.e. no need to squeeze the simp if followed by `omega`).